base/rational.jl

```diff
@@ -95,6 +95,22 @@ end
 function write(s::IO, z::Rational)
     write(s,numerator(z),denominator(z))
 end
+function parse(::Type{Rational{T}}, s::AbstractString) where T<:Integer  
+    ss = split(s, '/'; limit = 2)
+    if isone(length(ss))
+        return parse(T, only(ss)) // one(T) 
+    end    
+    @assert length(ss) == 2
+    ns, ds = ss
+    # Correctly account for non-ascii strings.
+    if first(ds, 1) == "/" 
```

Comment:
```suggestion
    if startswith(ds, '/')
```
or, equivalently, `first(ds) == '/'`. Your version is correct but may allocate a `String` (I don't know how smart the compiler is regarding this nowadays, but better to avoid it altogether).

Reply:
I think that `ds[1]` would also work (i.e., could not error) because it would the first character after `/`, which is ascii, so it cannot be part of a non-ascii character. But I'm not completely sure.

Reply:
I am also not completely sure about `ds[1]` because I do not know if the string cannot be encoded in a way that the first byte just happens to match `'/'`, `startswith` seems a sane approach, while I am a bit puzzled that the documentation does not mention anywhere that the second argument of `startswith` may be a single character.